integrationTests/mock/multiversXContractStateMock.go

```diff
@@ -169,36 +174,55 @@ func (mock *multiversXContractStateMock) createProposedTransfer(dataSplit []stri
 		BatchId: big.NewInt(0).SetBytes(buff),
 	}
 
-	for i := 2; i < len(dataSplit); i += 5 {
-		from, errDecode := hex.DecodeString(dataSplit[i])
+	currentIndex := 2
+	for currentIndex < len(dataSplit) {
+		from, errDecode := hex.DecodeString(dataSplit[currentIndex])
 		if errDecode != nil {
 			panic(errDecode)
 		}
 
-		to, errDecode := hex.DecodeString(dataSplit[i+1])
+		to, errDecode := hex.DecodeString(dataSplit[currentIndex+1])
 		if errDecode != nil {
 			panic(errDecode)
 		}
 
-		amountBytes, errDecode := hex.DecodeString(dataSplit[i+3])
+		amountBytes, errDecode := hex.DecodeString(dataSplit[currentIndex+3])
 		if errDecode != nil {
 			panic(errDecode)
 		}
 
-		nonceBytes, errDecode := hex.DecodeString(dataSplit[i+4])
+		nonceBytes, errDecode := hex.DecodeString(dataSplit[currentIndex+4])
 		if errDecode != nil {
 			panic(errDecode)
 		}
 
 		t := Transfer{
 			From:   from,
 			To:     to,
-			Token:  dataSplit[i+2],
+			Token:  dataSplit[currentIndex+2],
 			Amount: big.NewInt(0).SetBytes(amountBytes),
 			Nonce:  big.NewInt(0).SetBytes(nonceBytes),
 		}
 
+		indexIncrementValue := 5
+		if bytes.Equal(from, mock.whitelistedEthSCAddress) {
```

Comment:
The check should be done on `to` instead of `from` (for sc execution, is a transfer to a mvx smart contract) :D